medusa/providers/torrent/html/btdb.py

```diff
@@ -0,0 +1,157 @@
+# coding=utf-8
+
+"""Provider code for BTDB."""
+
+from __future__ import unicode_literals
+
+import logging
+
+from medusa import tv
+from medusa.bs4_parser import BS4Parser
+from medusa.helper.common import convert_size
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+from requests.compat import urljoin
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class BTDBProvider(TorrentProvider):
+    """BTDB Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(BTDBProvider, self).__init__('BTDB')
+
+        # Credentials
+        self.public = True
+
+        # URLs
+        self.url = 'https://btdb.to'
+        self.urls = {
+            'rss': urljoin(self.url, '/q/x264/?sort=time'),
+            'search': urljoin(self.url, '/q/{query}/{page}?sort=popular'),
+        }
+
+        # Miscellaneous Options
+        self.max_pages = 3
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=20)
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                    for page in range(1, self.max_pages + 1):
+                        search_url = self.urls['search'].format(query=search_string, page=page)
+    
+                        response = self.session.get(search_url)
+                        if not response or not response.text:
+                            log.debug('No data returned from provider')
+                            break
+    
+                        page_results = self.parse(response.text, mode)
+                        results += page_results
+                        if len(page_results) < 10:
+                            break
+    
+                    # We don't have the real seeds but we can sort results by popularity and
+                    # normalize seeds numbers so results can be sort in manual search
+                    results = self.calc_seeds(results)
+
+                else:
+                    response = self.session.get(self.urls['rss'])
+                    if not response or not response.text:
+                        log.debug('No data returned from provider')
+                        continue
+
+                    results += self.parse(response.text, mode)
```

Comment:
`Daily search` results should also use `self.calc_seeds(results)` as they are cached and will show up in the `Manual search` results.